wp-cli/civicrm.php

```diff
@@ -25,1253 +25,1305 @@
+			civicrm_setup( "$settings_dir" );
+			WP_CLI::launch( "chmod 0777 $settings_dir -R" );
+
+			# now we've got some files in place, require PEAR DB and check db setup
+			$dsn = "mysql://{$dbuser}:{$dbpass}@{$dbhost}/{$dbname}?new_link=true";
+			$dsn_nodb = "mysql://{$dbuser}:{$dbpass}@{$dbhost}";
+
+			if ( ! defined( 'DB_DSN_MODE' ) ) {
+				define( 'DB_DSN_MODE', 'auto' );
+			}
+
+			require_once "$crmPath/packages/DB.php";
+
+			$db = DB::connect( $dsn );
+			if ( DB::iserror( $db ) ) {
+				$db = DB::connect( $dsn_nodb );
+				if ( DB::iserror( $db ) ) {
+					return WP_CLI::error( 'Unable to connect to database. Please re-check credentials.' );
+				}
+				$db->query( "CREATE DATABASE $dbname" );
+				if ( DB::iserror( $db ) ) {
+					return WP_CLI::error( 'CiviCRM database was not found. Failed to create one.' );
+				}
+				$db->disconnect();
+			}
+
+			# install db
+			$sql_path = "$crmPath/sql";
+
+			# setup database with civicrm structure and data
+			WP_CLI::line( 'Loading CiviCRM database structure ..' );
+			civicrm_source( $dsn, $sql_path . '/civicrm.mysql' );
+			WP_CLI::line( 'Loading CiviCRM database with required data ..' );
+
+			# testing the translated sql files availability
+			$data_file = $sql_path . '/civicrm_data.mysql';
+			$acl_file  = $sql_path . '/civicrm_acl.mysql';
+
+			if ( '' != $lang ) {
+
+				if ( file_exists( $sql_path . '/civicrm_data.' . $lang . '.mysql' )
+					and file_exists( $sql_path . '/civicrm_acl.' . $lang . '.mysql' )
+					and '' != $lang
+				 ) {
+					$data_file = $sql_path . '/civicrm_data.' . $lang . '.mysql';
+					$acl_file = $sql_path . '/civicrm_acl.' . $lang . '.mysql';
+				} else {
+					WP_CLI::warning( "No sql files could be retrieved for '$lang' using default language." );
+				}
+			}
+
+			civicrm_source( $dsn, $data_file );
+			civicrm_source( $dsn, $acl_file );
+
+			WP_CLI::success( 'CiviCRM database loaded successfully.' );
+
+			# generate civicrm.settings.php file
+			$settings_tpl_file = "$crmPath/templates/CRM/common/civicrm.settings.php.tpl";
+			if ( ! file_exists( $settings_tpl_file ) ) {
+				$settings_tpl_file = "$crmPath/templates/CRM/common/civicrm.settings.php.template";
+				if ( ! file_exists( $settings_tpl_file ) ) {
+					return WP_CLI::error( 'Could not find CiviCRM settings template and therefore could not create settings file.' );
+				}
+			}
+
+			WP_CLI::line( 'Generating civicrm settings file ..' );
+
+			if ( $base_url = $this->getOption( 'site_url', false ) ) {
+				$ssl      = $this->getOption( 'ssl', false );
+				$protocol = ( 'on' == $ssl ? 'https' : 'http' );
+			}
+
+			$base_url = ! $base_url ? get_bloginfo( 'url' ) : $protocol . '://' . $base_url;
+			if ( substr( $base_url, -1 ) != '/' ) {
+				$base_url .= '/';
+			}
+			$params = array(
+				'crmRoot'            => $crmPath . '/',
+				'templateCompileDir' => "{$settings_dir}files/civicrm/templates_c",
```

Comment:
`'templateCompileDir' => "{$settings_dir}templates_c",`

Appears to be proper setting for 4.7